src/Features/CSharp/Portable/QuickInfo/CSharpSyntacticQuickInfoProvider.cs

```diff
@@ -125,5 +150,34 @@ private static bool TryFindFurthestNearbyComment<T>(ref T triviaSearchList, out
 
             return nearbyTrivia.IsSingleOrMultiLineComment();
         }
+
+        private static QuickInfoItem? BuildQuickInfoDirectives(SyntaxToken token, CancellationToken cancellationToken)
+        {
+            var directiveTrivia = token.Parent?.AncestorsAndSelf(ascendOutOfTrivia: false).FirstOrDefault(n =>
+                n is EndRegionDirectiveTriviaSyntax or ElifDirectiveTriviaSyntax or ElseDirectiveTriviaSyntax or EndIfDirectiveTriviaSyntax) as DirectiveTriviaSyntax;
+            if (directiveTrivia is EndRegionDirectiveTriviaSyntax endRegionDirectiveTrivia)
+            {
+                var regionStart = endRegionDirectiveTrivia.GetMatchingDirective(cancellationToken);
+                if (regionStart is not null)
+                {
+                    return QuickInfoItem.Create(token.Span, relatedSpans: ImmutableArray.Create(regionStart.Span));
+                }
+            }
+
+            if (directiveTrivia is ElifDirectiveTriviaSyntax or ElseDirectiveTriviaSyntax or EndIfDirectiveTriviaSyntax)
+            {
+                var matchingDirectives = directiveTrivia.GetMatchingConditionalDirectives(cancellationToken);
+                var lastMatchBefore = matchingDirectives
+                    .TakeWhile(d => d.SpanStart < directiveTrivia.SpanStart)
+                    .Select(d => d.Span)
```

Comment:
instead of having the quick info code have to add \r\n's as it's combining related spans, why not just include the \r\n portion in the span you're grabbing here.  tehn it should happen automatically, right?

Reply:
I tried this, but the newLine token got dismissed in the QuickInfo builder logic. I even tried to use the `sections` parameter but `TaggedText` does not support directives. See also https://github.com/dotnet/roslyn/pull/56609#issuecomment-927721391